library/std/src/sys/wasm/condvar_atomics.rs

```diff
@@ -42,13 +42,22 @@ impl Condvar {
 
     pub unsafe fn notify_one(&self) {
         self.cnt.fetch_add(1, SeqCst);
-        wasm32::memory_atomic_notify(self.ptr(), 1);
+        // SAFETY: ptr() is always valid
+        unsafe {
+            wasm32::atomic_notify(self.ptr(), 1);
+        }
     }
 
     #[inline]
     pub unsafe fn notify_all(&self) {
-        self.cnt.fetch_add(1, SeqCst);
-        wasm32::memory_atomic_notify(self.ptr(), u32::MAX); // -1 == "wake everyone"
+        // SAFETY: The caller must gurantee that the argument of `fetch_add()` is valid
+        // to describe memory ordering.
+        unsafe {
+            self.cnt.fetch_add(1, SeqCst);
+        }
```

Comment:
Not an unsafe operation (please also fix the other atomics).

Reply:
ok